### PR TITLE
Improve Android tests

### PR DESCRIPTION
**Pull request by descorp**
# Open PR

## Changes

* Improve tests
* Add `allowCreditCards` property to GooglePay configuration

### CODE REVIEW

File: android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt [hunk 1]
```diff
@@ -6,36 +6,52 @@ import org.junit.Test
...
+        Assert.assertFalse(analyticsParser.verboseLogs)
+    }
+
     @Test
     fun testIsEnabled() {
         // GIVEN
         val config = WritableMapMock()
         config.putBoolean(AnalyticsParser.ANALYTICS_ENABLED_KEY, true)
+        config.putBoolean(AnalyticsParser.ANALYTICS_VERBOSE_LOGS, false)
+
```
Comment (Robert-SD) [inline android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt:30]:
```suggestion
```

File: android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt [hunk 2]
```diff
@@ -6,36 +6,52 @@ import org.junit.Test
...
         val analyticsParser = AnalyticsParser(config)
 
         // THEN
-        Assert.assertSame(
-            analyticsParser.analytics.level,
-            AnalyticsLevel.ALL
-        )
+        Assert.assertSame(analyticsParser.analytics.level, AnalyticsLevel.ALL)
+        Assert.assertFalse(analyticsParser.verboseLogs)
+
```
Comment (Robert-SD) [inline android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt:38]:
```suggestion
```

File: android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt [hunk 3]
```diff
@@ -6,36 +6,52 @@ import org.junit.Test
...
         val analyticsParser = AnalyticsParser(config)
 
         // THEN
-        Assert.assertSame(
-            analyticsParser.analytics.level,
-            AnalyticsLevel.NONE
-        )
+        Assert.assertSame(analyticsParser.analytics.level, AnalyticsLevel.NONE)
+        Assert.assertTrue(analyticsParser.verboseLogs)
+
```
Comment (Robert-SD) [inline android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt:54]:
```suggestion
```

File: android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt [hunk 4]
```diff
@@ -6,36 +6,52 @@ import org.junit.Test
 
 class AnalyticsParserTest {
 
+    @Test
+    fun applyConfigurationOnSubDictionary() {
```
Comment (Robert-SD) [inline android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt:10]: The other tests start with "test...", shall we keep this naming convention here as well?

File: android/src/test/java/com/adyenreactnativesdk/configuration/GooglePayConfigurationParserTest.kt [hunk 5]
```diff
@@ -6,64 +6,77 @@
...
-        `when`(mockArray.toArrayList()).thenReturn(arrayListOf("PAN_ONLY", "CRYPTOGRAM_3DS"))
-        config.putArray(GooglePayConfigurationParser.ALLOWED_AUTH_METHODS_KEY, mockArray)
+        config.putBoolean(GooglePayConfigurationParser.ALLOW_PREPAID_CARDS_KEY, true)
+        config.putBoolean(GooglePayConfigurationParser.ALLOW_CREDIT_CARDS_KEY, true)
+        config.putBoolean(GooglePayConfigurationParser.EMAIL_REQUIRED_KEY, true)
+        config.putBoolean(GooglePayConfigurationParser.SHIPPING_ADDRESS_REQUIRED_KEY, true)
+        config.putBoolean(GooglePayConfigurationParser.BILLING_ADDRESS_REQUIRED_KEY, true)
+        config.putString(GooglePayConfigurationParser.TOTAL_PRICE_STATUS_KEY, "FINAL")
+        config.putString(GooglePayConfigurationParser.MERCHANT_ACCOUNT_KEY, "Merchant_account")
+        // TODO: add billing address tests
```
Comment (Robert-SD) [inline android/src/test/java/com/adyenreactnativesdk/configuration/GooglePayConfigurationParserTest.kt:56]: Shall the comments be removed?

File: android/src/test/java/com/adyenreactnativesdk/configuration/ThreeDSConfigurationParserTest.kt [hunk 6]
```diff
@@ -1,14 +1,34 @@
...
+        val threedsConfig = WritableMapMock()
+        config.putMap(ThreeDSConfigurationParser.THREEDS2_KEY, threedsConfig)
+
+        // WHEN
+        val sut = ThreeDSConfigurationParser(config)
+        sut.applyConfiguration(mockBuilder)
+
+        // THEN
+        verify(mockBuilder, times(0)).setThreeDSRequestorAppURL(any())
+
```
Comment (Robert-SD) [inline android/src/test/java/com/adyenreactnativesdk/configuration/ThreeDSConfigurationParserTest.kt:26]:
```suggestion
```

**Review by Robert-SD — APPROVED** [review]

File: android/src/test/java/com/adyenreactnativesdk/configuration/ThreeDSConfigurationParserTest.kt (hunk 6 above)
Comment (descorp) [reply android/src/test/java/com/adyenreactnativesdk/configuration/ThreeDSConfigurationParserTest.kt:26]: Will address formatting in another PR :)

File: android/src/test/java/com/adyenreactnativesdk/configuration/GooglePayConfigurationParserTest.kt (hunk 5 above)
Comment (descorp) [reply android/src/test/java/com/adyenreactnativesdk/configuration/GooglePayConfigurationParserTest.kt:56]: It is still missing 😅

File: android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt (hunk 2 above)
Comment (descorp) [reply android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt:38]: Will address formatting in another PR :)

File: android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt (hunk 1 above)
Comment (descorp) [reply android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt:30]: Will address formatting in another PR :)

File: android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt (hunk 3 above)
Comment (descorp) [reply android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt:54]: Will address formatting in another PR :)

File: android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt (hunk 4 above)
Comment (descorp) [reply android/src/test/java/com/adyenreactnativesdk/configuration/AnalyticsParserTest.kt:10]: Good one! Will address in next PR

**Pull request merged**